src/core/serialize/src/main/scala/params/UDFParam.scala

```diff
@@ -0,0 +1,17 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package org.apache.spark.ml.param
+
+import org.apache.spark.sql.expressions.UserDefinedFunction
+
+/** Param for UserDefinedFunction.  Needed as spark has explicit params for many different
+  * types but not UserDefinedFunction.
+  */
+class UDFParam(parent: Params, name: String, doc: String, isValid: UserDefinedFunction => Boolean)
+  extends ComplexParam[UserDefinedFunction](parent, name, doc, isValid) {
+
+  def this(parent: Params, name: String, doc: String) =
```

Comment:
make this a default arg in the class constructor?

Reply:
What would tat change look like? This is how PipelineStageParam works.